routers/api/v1/repo/pull.go

```diff
@@ -799,13 +799,25 @@ func MergePullRequest(ctx *context.APIContext) {
 		return
 	}
 
-	// set defaults to propagate needed fields
-	if err := form.SetDefaults(ctx, pr); err != nil {
-		ctx.ServerError("SetDefaults", fmt.Errorf("SetDefaults: %v", err))
-		return
+	if len(form.Do) == 0 {
+		form.Do = string(repo_model.MergeStyleMerge)
+	}
+
+	message := strings.TrimSpace(form.MergeTitleField)
```

Comment:
if form.SetDefaults is replaced by this we should move the whole **form.SetDefaults** into pull service and add the GetDefaultMergeMessage there